src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.RateLimiting;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Extension methods for the RateLimiting middleware options.
+/// </summary>
+public static class RateLimiterOptionsExtensions
+{
+    /// <summary>
+    /// Adds a new <see cref="TokenBucketRateLimiter"/> with the given <see cref="TokenBucketRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="RateLimiterOptions"/> to add a limiter to.</param>
+    /// <param name="policyName">The name that will be associated with the limiter.</param>
+    /// <param name="tokenBucketRateLimiterOptions">The <see cref="TokenBucketRateLimiterOptions"/> to be used for the limiter.</param>
+    /// <returns>This <see cref="RateLimiterOptions"/>.</returns>
+    public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions options, string policyName, TokenBucketRateLimiterOptions tokenBucketRateLimiterOptions)
```

Comment:
Do the parameters need null checks?

Reply:
`RateLimiterOptions` will throw if `policyName` is null, and `TokenBucketRateLimiter` will throw if `tokenBucketRateLimiterOptions` is null, which I think is fine